### PR TITLE
Register a shutdownhook to stop all the Kafka consumers

### PR DESCRIPTION
**Pull request by fx19880617**

### CODE REVIEW

File: pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/KafkaConsumerManager.java [hunk 1]
```diff
@@ -150,4 +150,30 @@ public void run() {
...
+    try {
+      // Shutdown all consumers
+      synchronized (KafkaConsumerManager.class) {
+        LOGGER.info("Trying to shutdown all the kafka consumers");
+        Iterator<ConsumerAndIterator> consumerIterator =
+            CONSUMER_AND_ITERATOR_FOR_CONFIG.values().iterator();
+
+        while (consumerIterator.hasNext()) {
+          ConsumerAndIterator consumerAndIterator = consumerIterator.next();
+          LOGGER.info("Trying to shutdown consumer/iterator {}", consumerAndIterator.getId());
```
Comment (kishoreg) [inline pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/KafkaConsumerManager.java]: what if we simply call the other method releaseConsumerAndIterator for each consumerAndIterator

**Comment by kishoreg** [comment]
LGTM.

File: pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/KafkaConsumerManager.java (hunk 1 above)
Comment (fx19880617) [inline pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/KafkaConsumerManager.java]: method releaseConsumerAndIterator will not close consumers currently been acquired or   consumers haven't been released for more than 1 min.

**Comment by jfim** [comment]
LGTM, I'll merge it later today. I'll amend the commit message to say "Adding method to stop all Kafka consumers," as there is no shutdown hook in this code.

**Pull request closed without merging**